File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        if (nums.length === 1) {
            return 1;
        }

        nums = nums.sort((a, b) => a - b);

        let maxLen = 0;
        let curLen = 1;

        for (let i = 0; i < nums.length - 1; i++) {
            if (nums[i+1] === nums[i] + 1) {
                curLen++;
            } else if (nums[i+1] !== nums[i]){
                curLen = 1;
            }

            if (curLen > maxLen) {
                maxLen = curLen;
            }
        }

        return maxLen;
    }
}

File: Data Structures & Algorithms/jump-game/submission-3.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        dp = [False] * n

        dp[n - 1] = True

        goal = n - 1

        for i in range(n - 2, -1, -1):
            if nums[i] >= goal - i:
                dp[i] = True
                goal = i
            
        
        return dp[0]
